src/ndarray/ndarray.cc

```diff
@@ -1711,7 +1711,7 @@ bool NDArray::Load(dmlc::Stream *strm) {
   // load shape
   mxnet::TShape shape;
   if (!shape.Load(strm)) return false;
-  if (shape.ndim() == 0) {
+  if (mxnet::op::shape_is_none(shape)) {
```

Comment:
We may need to check whether the current mode is numpy compatible. If not, we need to conver `shape` to numpy shape first and then call `shape_is_none/shape_is_known`.